Snap failing to build due to mdns-publisher

mdns-publisher is failing to build with the following error:

```
Cloning into '/root/parts/mdns-publisher/src'...
Fetching godeps...
cannot find package "golang.org/x/tools/go/vcs" in any of:
	/snap/go/8304/src/golang.org/x/tools/go/vcs (from $GOROOT)
	/root/parts/mdns-publisher/go/src/golang.org/x/tools/go/vcs (from $GOPATH)

go get -d github.com/rogpeppe/godeps
Failed to run 'go get -d github.com/rogpeppe/godeps' for 'mdns-publisher': Exited with code 1.
Verify that the part is using the correct parameters and try again.
Run the same command again with --debug to shell into the environment if you wish to introspect this failure.
Error: Process completed with exit code 2.
```

I detest that it's failing to build with zero changes from us. My proposal isn't to actually fix this: it's to **remove it entirely.**

So what does this component actually DO? It's broadcasts a `nextcloud.local` (or `<hostname>.local`) hostname via mdns, which allows for easy network discovery. It's a feature that has existed from the beginning, when the primary purpose of this snap was the Nextcloud Box, a hardware device based on Ubuntu Core, which was entirely headless. Using mdns meant that you didn't actually need to connect a monitor to it or otherwise determine its IP address before being able to access it. That worked reasonably well, but today I think it's an out-of-date concern. The Nextcloud Box no longer exists, and thus the use-case behind why this component exists in the first place no longer exists.

The downside is: I don't generally like removing things from the snap. It's entirely possible there are users out there that depend on this feature. But that seems unlikely at this point. And if we get bugs logged by people who need it, we can always solve the build failure and add it back in. It's a scream test, which I hate, but it seems warranted here given that this isn't the first time this component has suddenly ceased building.

Thoughts?

--- a/README.md
+++ b/README.md
@@ -7,7 +7,6 @@ Nextcloud server packaged as a snap. It consists of:
 - PHP 8.0
 - MySQL 8
 - Redis 6
-- mDNS for network discovery
 
 
 ## How to install
@@ -19,12 +18,6 @@ stable one, but you may be interested in others.
 
 
 ## How to use
-
-After install, assuming you and the device on which it was installed are on the
-same network, you should be able to reach the Nextcloud installation by visiting
-`<hostname>.local` in your browser. If your hostname is `localhost` or
-`localhost.localdomain`, like on an Ubuntu Core device, `nextcloud.local` will
-be used instead.
 
 Upon visiting the Nextcloud installation for the first time, you'll be prompted
 for an admin username and password. After you provide that information you'll be

--- a/snap/snapcraft.yaml
+++ b/snap/snapcraft.yaml
@@ -58,13 +58,6 @@ apps:
     restart-condition: always
     plugs: [network, network-bind]
 
-  # mDNS daemon
-  mdns-publisher:
-    command: bin/delay-on-failure mdns-publisher nextcloud
-    daemon: simple
-    restart-condition: always
-    plugs: [network, network-bind]
-
   # MySQL client
   mysql-client:
     command: bin/run-mysql
@@ -412,11 +405,6 @@ parts:
     plugin: dump
     prime:
       - -*
-
-  mdns-publisher:
-    plugin: godeps
-    source: https://github.com/kyrofa/mdns-publisher.git
-    go-importpath: github.com/kyrofa/mdns-publisher
 
   delay-on-failure:
     plugin: dump

